addons/binding/org.openhab.binding.freemobilesms/src/main/java/org/openhab/binding/freemobilesms/internal/FreeMobileSmsHandler.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project.
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.freemobilesms.internal;
+
+import static org.openhab.binding.freemobilesms.internal.FreeMobileSmsBindingConstants.*;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.freemobilesms.internal.FreeMobileSmsConfiguration;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import fr.free.smsapi.Account;
+import fr.free.smsapi.Sender;
+import fr.free.smsapi.RawAccount;
+
+/**
+ * The {@link FreeMobileSmsHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Guilhem Bonnefille <guilhem.bonnefille@gmail.com> - Initial contribution
+ */
+@NonNullByDefault
+public class FreeMobileSmsHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FreeMobileSmsHandler.class);
+
+    private FreeMobileSmsConfiguration config;
+
+    private final Sender sender = new Sender();
+
+    private Account account;
+
+    public FreeMobileSmsHandler(Thing thing) {
+        super(thing);
+    }
+
+    protected void sendMessage(String message) {
+      logger.debug("Message to send: {}", message);
+      try {
+        sender.send(account, message);
+      } catch (IOException e) {
+        logger.error("Failed to send message: {}", e.getMessage());
+      }
+    }
+
+    private void handleMessage(Command command) {
+        logger.debug("Message command");
+        if (command instanceof StringType) {
+            sendMessage(command.toString());
+        } else {
+            logger.warn("Unsupported Command type");
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (CHANNEL_MESSAGE.equals(channelUID.getId())) {
```

Comment:
Exit early paradigm. So invert the condition and return:
```java
if (!CHANNEL_MESSAGE.equals(channelUID.getId())) {
   logger.warn("");
   return;
}
```

Reply:
Refactored